src/transformers/models/gptj/modeling_gptj.py

```diff
@@ -319,7 +332,9 @@ def forward(
         outputs = attn_outputs[1:]
 
         feed_forward_hidden_states = self.mlp(hidden_states)
-        hidden_states = attn_output + feed_forward_hidden_states + residual
+        hidden_states = self.attention_adapters(attn_output, residual, None)
```

Comment:
I believe the call to `attention_adapters()` should be placed before the forward pass through `mlp()` here. (Similar to the implementation in `GPT2Block` in `modeling_gpt2.py`.

Reply:
I see, compared with other Transformer implementations, GPT-J doesn't place a separate residual around the attention block. What I was concerned about here was that now, in contrast to the typical adapter placement, the MLP layer does not use the outputs of the attention adapter.

Reply:
Correct. That is why it is different in GPT-J

Reply:
We'll keep it as you suggested then